Fix the Helm exporter blogpost

zeenix opened the pull request:
I'm not so sure of the second post, it removes the step of `helm dep build` but introduces the step of `helm init -c` (i-e two invocations of `helm init` in the same doc). We could avoid that if we move the instructions for setting up helm before exporter command but that's a big change and I'm not sure if Chef folks will be good with that.

Commit:
Remove Habitat Operator steps from Helm blog post

Some of the steps were not needed and rest became redundant with Habitat
0.54.

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -22,28 +22,15 @@ And now, let's get started!
 So, for the purposes of this Helm demo to work locally, you now need to do: 
 
-* Be in the same directory as your compiled habitat-operator (i.e. habitat-operator)
 * Start minikube with RBAC enabled: `minikube start --extra-config=apiserver.Authorization.Mode=RBAC` 
```
Considering the operator requires certain k8s version to work, I would pass in a specific k8s version. Either `1.8.4` or `1.9.0` for example.

zeenix replied:
pass how? I'm not a minikube guy :)

lilic replied:
either:
`minikube start --kubernetes-version v1.8.0 --extra-config=apiserver.Authorization.Mode=RBAC`

or:
`minikube start --kubernetes-version v1.9.0`

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -22,28 +22,15 @@ And now, let's get started!
-* Be in the same directory as your compiled habitat-operator (i.e. habitat-operator)
 * Start minikube with RBAC enabled: `minikube start --extra-config=apiserver.Authorization.Mode=RBAC` 
 * Tell your machine's Docker daemon to use minikube's instead: `eval $(minikube docker-env)`
 
```
Can you maybe replace the following with the actual command a user would need to use? 

> and use `hab pkg export helm --help` to see more available options.

zeenix replied:
Huh? the above step in this context doens't match your comment. Maybe it was supposed to be added in another part of the file?

lilic replied:
It does, just expand, I can't comment on the not changed parts of the file :) 
So the following:
>  If you would prefer to publish your Helm charts to your Dockerhub and have Helm find them there to use on your Kubernetes cluster, that is supported! Enter the Habitat Studio (`hab studio enter`) and use `hab pkg export helm --help` to see more available options.

zeenix replied:
ah ok, yeah.

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +44,24 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
 * Enter the Habitat Studio within the nginx core plan with `hab studio enter`
 * Build the latest nginx package and stores it as a habitat artifact (“hart”) in the nginx plan’s `results` directory: `build`
-* Export the hab artifact to a Helm chart: `hab pkg export helm results/<yourfilename>.hart` This command will look very similar to `hab pkg export helm results/tdrew-nginx-1.11.10-20180221053714-x86_64-linux.hart`
+* Initialize the Helm client-side: `helm init -c`
```
Is this really needed?

zeenix replied:
As I explained in the PR description, with the current layout, yes. The question is if we want to make big changes (moving sections and add new sections) to this post?

lilic replied:
Feel free to ask Tasha on the upstream PR?

zeenix replied:
Sure.

lilic replied:
I am a bit confused by this, can you even do helm while inside the studio?

zeenix replied:
I'm not sure but I will try once I can get the studio working here.

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +44,24 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
 * Build the latest nginx package and stores it as a habitat artifact (“hart”) in the nginx plan’s `results` directory: `build`
-* Export the hab artifact to a Helm chart: `hab pkg export helm results/<yourfilename>.hart` This command will look very similar to `hab pkg export helm results/tdrew-nginx-1.11.10-20180221053714-x86_64-linux.hart`
+* Initialize the Helm client-side: `helm init -c`
+* Export the hab artifact to a Helm chart: `hab pkg export helm --download-deps results/<yourfilename>.hart` This command will look very similar to `hab pkg export helm --download-deps results/tdrew-nginx-1.11.10-20180221053714-x86_64-linux.hart`
```
I would put a new line here before the commands, to make it clearer.

zeenix replied:
on a separate para you mean?

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +44,24 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
 ### Start a cluster and deploy your helm chart!
 
-Here we are going to set up your minikube, and then deploy our helm chart onto that cluster. So, go back to your `habitat-operator` directory where you left your minikube cluster running and: 
+Here we are going to set up your minikube, and then deploy our helm chart onto that cluster.
```
minikube?

zeenix replied:
yeah, i was also wondering.

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +44,24 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
-* `helm dependency build <what your pwd was from building your Helm + Habitat package>` Reads your new nginx-<version>-<timestamp>/requirements.yaml file and sees the dependency on the habitat-operator. Pulls down the chart that describes the Habitat Operator and embeds it in your application’s chart 
 * `helm install /Users/tdrew/core-plans/nginx/results/nginx-1.11.10-20180220023948/` (or whatever your pwd was -- if it's exactly this we need to talk.) Has Helm install your Habitat Helm package on your Minikube cluster
 
 You will now see output similar to this: 
```
Please change this to have a status `1/1` pods. :)

zeenix replied:
will do, thanks.

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +44,24 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
-* `kubectl create -f examples/rbac` This gives the Habitat Operator the permissions it needs on your RBAC-enabled cluster
-* `kubectl apply -f examples/rbac/habitat-operator.yml` Deploys the Habitat Operator in your minikube cluster 
+* `kubectl apply -f https://raw.githubusercontent.com/kinvolk/habitat-operator/v0.5.1/examples/rbac/habitat-operator.yml` This gives the Habitat Operator the permissions it needs on your RBAC-enabled cluster
+* `kubectl apply -f https://raw.githubusercontent.com/kinvolk/habitat-operator/v0.5.1/examples/rbac/rbac.yml` Deploys the Habitat Operator in your minikube cluster
```
So we can use `v0.5.1` now? If your changes were merged, we don't need to deploy this as helm exporter will do that no?

zeenix replied:
ah yes. These are now redundant.

Commit:
Simplify `hab pkg export helm --help` instruction

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

Commit:
Put the Helm export commands on separate paras

It's slightly easier to follow and read that way.

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

Commit:
Fix description of a step in Helm exporter blog post

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

Commit:
Show pod as ready in the Helm blog post

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

zeenix commented:
PTALA.

Commit:
Remove redundant Tiller setup step

Tiller is setup correctly with RBAC later in the post so it's best not to
ask user to set it up twice, each time with different configuration.

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -22,32 +22,18 @@ And now, let's get started!
 * Habitat (`brew install habitat`) (check with `hab --version` to make sure you're on 0.54.0) Or, if you prefer the `curl | bash`, try `curl https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh | sudo bash`
```
Nit:
Above here it says:
>As of Habitat 0.54.0, you can export your Habitat built applications into Helm!

I think that was sooner no?

lilic replied:
Nit pick:
> Habitat (brew install habitat) (check with hab --version to make sure you're on 0.54.0)
I would change to
>Habitat (brew install habitat) (check with hab --version to make sure you're on 0.54.0 or above)
or something like that?

zeenix replied:
No, these instructions won't work for 0.53 for two reasons:

* operator setup not being taken care of.
* version number in the chart dir name

lilic replied:
Yes, was just a nit, I understand that, I meant that Helm exporter was introduced earlier already.... hence the nitpicking part above :)

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -22,32 +22,18 @@ And now, let's get started!
 
 **Important Note** 
 
 Minikube uses its own Docker Engine, not your machine's Docker Engine. Habitat Studio uses your machine's Docker Engine. So, in order for your Minikube do be able to find the Helm package that you are about to build and export, you need to point your machine's Docker Engine to use Minikube's instead (once you've started your Minikube cluster). 
```
> Minikube do be able to find the Helm package
instead do:
> Minikube do be able to find the exported image

zeenix replied:
it follows with "that you are about to build and export" so not sure that's needed but we do need to fix the typo "do be" -> "to be".

lilic replied:
But this is not needed for minikube to find the helm package, but the exported docker image that will be "pushed" to your docker.

Did not notice do be part! 👍

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -22,32 +22,18 @@ And now, let's get started!
 
-* Be in the same directory as your compiled habitat-operator (i.e. habitat-operator)
-* Start minikube with RBAC enabled: `minikube start --extra-config=apiserver.Authorization.Mode=RBAC` 
+* Start minikube with RBAC enabled: `minikube start --kubernetes-version v1.9.0`
```
Start a local Kubernetes RBAC enabled cluster using minikube e.g.: `minikube start --kubernetes-version v1.9.0`

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -22,32 +22,18 @@ And now, let's get started!
 * Tell your machine's Docker daemon to use minikube's instead: `eval $(minikube docker-env)`
 
-If you would prefer to publish your Helm charts to your Dockerhub and have Helm find them there to use on your Kubernetes cluster, that is supported! Enter the Habitat Studio (`hab studio enter`) and use `hab pkg export helm --help` to see more available options.
+If you would prefer to publish your Helm charts to your Dockerhub and have Helm find them there to use on your Kubernetes cluster, that is supported! Please run `hab pkg export helm --help` to see more available options.
```
I meant here provide detailed instructions how to push the image? eg: `hab pkg export helm --user zeeshan --password mycoolpassword`

zeenix replied:
that's what I wanted to do when we sent the instructions to Tasha but @blixtra said not to bother and then Tasha didn't seem to bother so I don't want to bother either anymore. :)

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +43,29 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
+
 * Exit the Habitat Studio: `exit`
 
 You now have a Helm chart for core-plans/nginx in your core-plans/nginx/nginx-<version>-<datestamp>/ directory
```
Think this para is now a leftover from before when we were building the operator. Can be removed I think?

zeenix replied:
Good catch!

zeenix replied:
actually no! This is about the helm chart dir that was just generated above.

lilic replied:
I am sorry. Meant to be placed below this:
>  I suggest you pwd for the above ^^ so you can use it in another tab when you have started your cluster and want to know where your helm chart lives

The talk about needing to use another tab...

zeenix replied:
I'm not sure this is related to that since the instruction didn't actually launch operator and the para isn't even talking of the operator. Am I missing something?

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -57,24 +43,29 @@ Once you've cloned core-plans locally, go to `core-plans/nginx` and enter the Ha
 * `kubectl -n kube-system create sa tiller` Creates a service account for Tiller in the kube-system namespace
 * `kubectl create clusterrolebinding tiller --clusterrole cluster-admin --serviceaccount=kube-system:tiller` Creates a ClusterRoleBinding for Tiller
 * `helm init --service-account tiller` Installs Tiller, specifying new service account 
 * `helm version` Check and make sure you have the right version of helm running on your client & server side, (as of printing, v 2.8.1) 
```
Should this not be before the previous instruction?

zeenix replied:
yeah

lilic commented on www/source/blog/2018-02-21-Habitat-Helm.html.md:
```diff
@@ -106,7 +97,7 @@ nginx-1.11.10-20180221053714  0s
 ==> v1/Pod(related)
 NAME                                             READY  STATUS             RESTARTS  AGE
-honorary-wolf-habitat-operator-7f74859c6d-2jv9b  0/1    ContainerCreating  0         0s
+honorary-wolf-habitat-operator-7f74859c6d-2jv9b  1/1    ContainerCreating  0         0s
```
When it's 1/1 the status is no `ContainerCreating` but `Running`.

Also think this should be nginx?

zeenix replied:
no it really does use the `habitat-operator`, that's one issue with helm that I don't understand.

Commit:
Fix a typo in the Helm exporter blog post

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

tashimi committed:
Add habitat-operator install step

Signed-off-by: Tasha Drew <tasha.drew@gmail.com>

thesentinels committed:
Merge pull request #4649 from habitat-sh/tashimi-patch-3

Approved by: @Nobody from Nowhere
Merged by: The Sentinels

Commit:
Fix instructions for Helm deps setup

Signed-off-by: Zeeshan Ali <zeeshan@kinvolk.io>

tashimi committed:
Update command

Signed-off-by: Tasha Drew <tasha.drew@gmail.com>
